sdk/python/feast/infra/aws_dynamo_provider.py

```diff
@@ -0,0 +1,201 @@
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import boto3
+import mmh3
+import pandas
+from botocore.exceptions import ClientError
+
+from feast import FeatureTable, utils
+from feast.feature_view import FeatureView
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.offline_stores.helpers import get_offline_store_from_sources
+from feast.infra.provider import (
+    Provider,
+    RetrievalJob,
+    _convert_arrow_to_proto,
+    _get_column_names,
+    _run_field_mapping,
+)
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.registry import Registry
+from feast.repo_config import DynamoOnlineStoreConfig, RepoConfig
+
+
+class AwsDynamoProvider(Provider):
+    def __init__(self, config: RepoConfig):
+        assert isinstance(config.online_store, DynamoOnlineStoreConfig)
+
+    def _initialize_dynamodb(self):
+        return boto3.resource("dynamodb")
+
+    def update_infra(
+        self,
+        project: str,
+        tables_to_delete: Sequence[Union[FeatureTable, FeatureView]],
+        tables_to_keep: Sequence[Union[FeatureTable, FeatureView]],
+        partial: bool,
+    ):
+        dynamodb = self._initialize_dynamodb()
+
+        for table_name in tables_to_keep:
+            table = None
+            try:
+                table = dynamodb.create_table(
+                    TableName=table_name.name,
+                    KeySchema=[
+                        {"AttributeName": "Row", "KeyType": "HASH"},
+                        {"AttributeName": "Project", "KeyType": "RANGE"},
+                    ],
+                    AttributeDefinitions=[
+                        {"AttributeName": "Row", "AttributeType": "S"},
+                        {"AttributeName": "Project", "AttributeType": "S"},
+                    ],
+                    ProvisionedThroughput={
+                        "ReadCapacityUnits": 5,
+                        "WriteCapacityUnits": 5,
```

Comment:
Probably it would be good to set this number as a parameter of dynamodb online store.